woocommerce/woocommerce
# Remove old in-app marketplace pages and redirect to new in-app section

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





After introducing new in-app marketplace, the older marketplace pages were not removed from WooCommerce Core. Even if the menu is hidden these pages are still accessible.

This PR removes the old in-app screens and add redirects to requests for old in-app pages.

#### Redirects:

|| Old in-app Marketplace URL  | Redirected to |
|---------------| ------------- | ------------- |
|Search| `/wp-admin/admin.php?page=wc-addons&section=search`  | `/wp-admin/admin.php?page=wc-admin&tab=extensions&path=%2Fextensions`  |
|My Subscriptions| `/wp-admin/admin.php?page=wc-addons&section=helper`  | `/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions`  |
|Discover| `/wp-admin/admin.php?page=wc-addons&section=_featured` | `/wp-admin/admin.php?page=wc-admin&path=%2Fextensions` |
|Categories|`/wp-admin/admin.php?page=wc-addons&section=free-extensions`|`/wp-admin/admin.php?page=wc-admin&tab=extensions&path=%2Fextensions&category=free-extensions`|
|Search Terms|`wp-admin/admin.php?search=shipping&page=wc-addons&section=_all` or `wp-admin/admin.php?search=shipping&page=wc-addons`|`/wp-admin/admin.php?page=wc-admin&term=mail&tab=search&path=%2Fextensions`|


Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Testing Redirects
1. Checkout this branch
2. Build assets. Run `pnpm --filter='@woocommerce/plugin-woocommerce' build` from repo root directory 
3. Start the WCCore test installtion: `pnpm -- wp-env start`  and test above redirects in http://localhost:8888/
4. Redirects for My Subscriptions, Search and Discover (featured) pages should be working as expected.
5. Make sure requests to old marketplace with search terms (e.g: `http://localhost:8888/wp-admin/admin.php?search=shipping&page=wc-addons&section=_all`) get redirected to new in-app search with correct search term.
6. Make sure requests to old in-app pages with categories are redirected to correct category in new in-app marketplace. Invalid categories gets redirected to Discover tab.

#### Testing WCCOM connect/disconnect
- Create a build from this PR or use this 
[woocommerce.zip](https://github.com/woocommerce/woocommerce/files/15274002/woocommerce.zip)
- Upload this to a jurassic.ninja site or another site for testing.
- Test connecting and disconnecting the site via https://woocommerce.com/my-account/my-stores/
- You should get redirected to new in-app market place for connection and disconnection.
- Connection/disconnection should work as expected.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Redirect old market-place to new market-place pages

#### Comment 

</details>
